program/tests/duplicate_block_proof.rs

```diff
@@ -254,6 +272,41 @@ async fn valid_proof_data() {
         .process_transaction(transaction)
         .await
         .unwrap();
+
+    // Verify that the report was written
+    let (report_account, _) = Pubkey::find_program_address(
+        &[
+            &leader.pubkey().to_bytes(),
+            &slot.to_le_bytes(),
+            &[u8::from(ProofType::DuplicateBlockProof)],
+        ],
+        &spl_slashing::id(),
+    );
+    let report_account = context
+        .banks_client
+        .get_account(report_account)
+        .await
+        .unwrap()
+        .unwrap();
+    let violation_report_size = std::mem::size_of::<ViolationReport>();
+    let violation_report: &ViolationReport =
+        pod_from_bytes(&report_account.data[0..violation_report_size]).unwrap();
+    assert_eq!(violation_report.reporter, reporter);
+    assert_eq!(violation_report.destination, destination);
+    assert_eq!(u64::from(violation_report.epoch), EPOCH);
+    assert_eq!(violation_report.pubkey, leader.pubkey());
+    assert_eq!(u64::from(violation_report.slot), slot);
+    assert_eq!(
+        ProofType::from(violation_report.violation_type),
+        ProofType::DuplicateBlockProof
+    );
+    assert_eq!(violation_report.proof_account, account.pubkey());
```

Comment:
nit: it might be easier to compare it to a created `ViolationReport`, since it implements `PartialEq`, and that way the compiler will complain if we add any fields to it later and forget to check them, ie

```
let expected_violation_report = ViolationReport { ... };
assert_eq!(violation_report, expected_violation_report);
```

Reply:
Yeah good point that's cleaner, 69c5f59731b0041d8c9e4a96937aae20b7e169ae